app/screens/buyers/BuyingView.java

```diff
@@ -2,37 +2,60 @@
 
 import icebase.app.App;
 
+import java.util.ArrayList;
 import java.util.Scanner;
 
 import icebase.app.MenuTitle;
+import icebase.app.Router;
+import icebase.app.enums.CATEGORY_ENUM;
+import icebase.app.enums.SCREEN_ENUM;
 import icebase.app.screens.Screen;
+import icebase.app.types.Store;
 
 public class BuyingView implements Screen {
+    private String storeName;
+    private Store store;
+    private Scanner sc = App.sc;
+    private ArrayList<String> categoryNames = new ArrayList<>();
+    private int choice;
+
+    public BuyingView(Store store) {
+        this.store = store;
+        this.storeName = store.getName();
+        for (CATEGORY_ENUM categoryENUM : CATEGORY_ENUM.values()) {
+            categoryNames.add(categoryENUM.value);
+        }
+        categoryNames.add("return");
+    }
 
     public void display() {
-        Scanner sc = App.sc;
-        int choice;
-        String storeName = "STORE X"; // stand-in value, should be set from constructor
-        String[] categoryList = { "Category A", "Category B", "Category C", "Return" }; // stand-in list
-        int range = categoryList.length;
+        CATEGORY_ENUM category;
 
         while (true) {
             MenuTitle.displayStoreName(storeName);
-            MenuTitle.displaySubTitle("[CATEGORIES]");
-            // MenuTitle.displayOptions(categoryList);
+            MenuTitle.displaySubTitle("CATEGORIES");
+            MenuTitle.displayOptions(categoryNames);
+            getChoice(); // gets and validates the choice given by the user
+            if (choice == categoryNames.size()) {
+                return;
+            }
+            category = CATEGORY_ENUM.values()[choice - 1];
+            Router.nagivate(SCREEN_ENUM.BUYING_ITEM_LIST, store, category);
+        }
+    }
 
+    private void getChoice() {
```

Comment:
also same comment nga mag return nalang siya dapat ng value,

hmmm it seems nga ga use tayo ng this function across different screens, i think we can extract to a helper Class?

maybe create a folder within `app` nga `helpers`

tapos name the class `ChoiceHelper.java`

tapos it just accepts the size sa options mo

Reply:
pwede² para sad nga uniform ang pag handle sa choices since ako na notice nga lain² amo ginabuhat